python-avd/pyavd/_cv/workflows/finalize_workspace_on_cv.py

```diff
@@ -87,3 +101,157 @@ async def finalize_workspace_on_cv(workspace: CVWorkspace, cv_client: CVClient)
+                        )
+                    ],
+                    warnings=[
+                        CVWorkspaceBuildImageValidationWarning(
+                            sku=getattr(image_validation_warning, "sku", None),
+                            warning_code=getattr(image_validation_warning, "warning_code", None),
+                            warning_msg=getattr(image_validation_warning, "warning_msg", None),
+                        )
+                        for image_validation_warning in getattr(
+                            getattr(getattr(getattr(response, "value", None), "image_validation_result", None), "warnings", None), "values", []
+                        )
+                        if not any(
+                            pattern.search(getattr(image_validation_warning, "warning_msg", "")) for pattern in compiled_workspace_build_warning_suppress_list
+                        )
+                    ]
+                    if workspace.build_warnings
+                    else [],
+                ),
+            )
+            async for response in workspace_build_details_stream_response
+            if (
+                getattr(getattr(getattr(response, "value", None), "config_validation_result", None), "errors", None)
+                or getattr(getattr(getattr(response, "value", None), "image_validation_result", None), "errors", None)
+            )
+            or (
+                workspace.build_warnings
+                and (
+                    getattr(getattr(getattr(response, "value", None), "config_validation_result", None), "warnings", None)
+                    or getattr(getattr(getattr(response, "value", None), "image_validation_result", None), "warnings", None)
+                )
+                and (
+                    len(compiled_workspace_build_warning_suppress_list) == 0
+                    or {False}
+                    in [
+                        {bool(pattern.search(i.error_msg)) for pattern in compiled_workspace_build_warning_suppress_list}
+                        for i in getattr(getattr(getattr(getattr(response, "value", None), "config_validation_result", None), "warnings", None), "values", [])
+                    ]
+                    or {False}
+                    in [
+                        {bool(pattern.search(i.warning_msg)) for pattern in compiled_workspace_build_warning_suppress_list}
+                        for i in getattr(getattr(getattr(getattr(response, "value", None), "image_validation_result", None), "warnings", None), "values", [])
+                    ]
+                )
+            )
+        ]
+
+    except Exception as e:
+        LOGGER.warning(
+            "process_workspace_build_details: Failed to fetch and process Workspace build results for workspace '%s' and build_id '%s'. Error: '%s'",
+            workspace.id,
+            workspace.build_id,
+            e,
+        )
+        return []
```

Comment:
Let's to this try/except in the calling function instead of wrapping the full code here, making it more indented and harder to read.

Reply:
Or even better make sure to try/except only the bits where you expect an issue that cannot be caught in a better way.

Reply:
Fixed. Removed that original large try/except as the only embedded call 'cv_client.get_workspace_build_details' already has try/except in it and the rest of the code is just a processing of the vars